src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const sepatareDate = date.split(fromFormat[3]);
+  const newDate = [];
+  const oldYearIndex = fromFormat.indexOf('YY');
+  const separator = toFormat[3];
+
+  if (sepatareDate[oldYearIndex] >= 30) {
+    sepatareDate[oldYearIndex] = '19' + sepatareDate[oldYearIndex];
+  }
+
+  if (sepatareDate[oldYearIndex] < 30) {
+    sepatareDate[oldYearIndex] = '20' + sepatareDate[oldYearIndex];
+  }
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    let index = toFormat.indexOf(fromFormat[i]);
+
+    if (index === -1) {
+      if (fromFormat[i] === 'YYYY') {
+        index = toFormat.indexOf('YY');
+
+        sepatareDate[i] = sepatareDate[i].slice(2);
```

Comment:
This line assumes that `sepatareDate[i]` is always a 4-digit year when `fromFormat[i]` is 'YYYY'. Ensure that this assumption holds true based on the input format.